src/crystals/crystals.controller.ts

```diff
@@ -0,0 +1,82 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Logger,
+  Post,
+  Query,
+  UseGuards,
+} from '@nestjs/common';
+import { ApiOperation, ApiResponse, ApiTags } from '@nestjs/swagger';
+import { AuthGuard } from '../auth/auth.guard';
+import { JwtProcessorType } from '../auth/auth.service';
+import { JwtType } from '../auth/jwt/jwt.type.decorator';
+import { CreateCrystalRequest } from './api/CreateCrystalRequest';
+import { CrystalDto } from './api/CrystalDto';
+import { CrystalsService } from './crystals.service';
+
+@Controller('/api/crystals')
+@ApiTags('crystals controller')
+export class CrystalsController {
+  private readonly logger = new Logger(CrystalsController.name);
+
+  constructor(private readonly crystalsService: CrystalsService) {}
+
+  @UseGuards(AuthGuard)
+  @JwtType(JwtProcessorType.RSA)
+  @Post()
+  @ApiOperation({
+    description: 'creates crystal',
+  })
+  @ApiResponse({
+    type: CrystalDto,
+    status: 200,
+  })
+  async createCrystal(
+    @Body() req: CreateCrystalRequest,
+  ): Promise<CreateCrystalRequest> {
+    this.logger.debug('Create crystal.');
+    return CrystalDto.covertToApi(
+      await this.crystalsService.createCrystal(
+        req.name,
+        req.category,
+        req.photo_URL,
+        req.short_description,
+      ),
+    );
+  }
```

Comment:
We don't provide the ability to create products via API. You should remove this entry point